### PR TITLE
Review view rework 3

### PR DESCRIPTION
**Pull request by thet**

### CODE REVIEW

File: src/recensio/plone/browser/review.py [hunk 1]
```diff
@@ -229,15 +119,17 @@ def get_doi_url_if_active(self):
     def get_metadata(self):  # noqa: C901
         context = self.context
         meta = {}
-        fields = []  # XXX
-        for field in self.metadata_fields:
+
+        for field in self.context.metadata_fields:
```
Comment (ale-rt) [inline src/recensio/plone/browser/review.py:123]: Please keep the `metadata_fields` in the view. That attribute is not used in any other place and has to stay in the view

**Review by ale-rt — CHANGES_REQUESTED** [review]

File: src/recensio/plone/browser/review.py [hunk 2]
```diff
@@ -229,15 +119,17 @@ def get_doi_url_if_active(self):
...
             value = False  # A field is only displayed if it has a value
             is_macro = False
-            if field.startswith("get_"):
-                label = self.custom_metadata_field_labels[field]
```
Comment (ale-rt) [inline src/recensio/plone/browser/review.py:237]: Why was custom_metadata_field_labels removed?

Comment (thet) [reply src/recensio/plone/browser/review.py:237]: because in `metadata_fields` there was only `get_journal_title` defined which was in fact `get_presentation_title`. no other `custom_metadata_field_labels` were used.

File: src/recensio/plone/browser/review.py (hunk 1 above)
Comment (thet) [reply src/recensio/plone/browser/review.py:123]: i wonder if we need this at all.
but for now i can move these definitions into the view.

File: src/recensio/plone/browser/review.py [hunk 3]
```diff
@@ -41,6 +34,23 @@ class View(BrowserView, CanonicalURLHelper):
...
 
+    _fields = None
+
+    @property
+    def fields(self):
+        if self._fields is None:
+            self._fields = {}
+            schemata = iterSchemata(self.context)
+            for schema in schemata:
+                for attr in schema.names:
```
Comment (ale-rt) [inline src/recensio/plone/browser/review.py:45]: This is a method
```suggestion
                for attr in schema.names():
```

File: src/recensio/plone/browser/review.py [hunk 4]
```diff
@@ -41,6 +34,23 @@ class View(BrowserView, CanonicalURLHelper):
...
+                    self._fields[attr] = schema.get(attr)
+        return self._fields
+
+    def get_label(self, field):
+        """Return the metadata label for a field of a particular
+        portal_type."""
+        return self.fields[field].label
```
Comment (ale-rt) [inline src/recensio/plone/browser/review.py:52]: Maybe:
```suggestion
        return self.fields[field].title
```

File: src/recensio/plone/browser/templates/review.pt [hunk 5]
```diff
@@ -154,7 +154,7 @@
       </div>
 
       <div tal:define="
-             metadata_field_names view/metadata_fields;
+             metadata_field_names context/metadata_fields;
```
Comment (ale-rt) [inline src/recensio/plone/browser/templates/review.pt:157]: Please revert this change

Comment (thet) [reply src/recensio/plone/browser/templates/review.pt]: :+1:

**Review by ale-rt — APPROVED** [review]
Good enough
